src/SumoCoders/FrameworkCoreBundle/Form/Extension/DateTypeExtension.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Form\Extension;
+
+use SumoCoders\FrameworkCoreBundle\Exception\MissingOptionException;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+final class DateTypeExtension extends AbstractTypeExtension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getExtendedType()
+    {
+        return 'date';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(
+            array(
+                'date_type' => 'normal',
+                'maximum_date' => null,
+                'minimum_date' => null,
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['date_type'] = $options['date_type'];
+
+        switch ($options['date_type']) {
+            case 'start':
+                if (!isset($options['minimum_date'])) {
+                    throw new MissingOptionException('minimum_date is missing');
```

Comment:
Bestaat er geen dergelijke exception van het symfony form component zelf? Zij doen daar toch ook validatie op? Als dat zo is zou ik dat type exception gebruiken, ipv een custom exception te maken.